### PR TITLE
Feature/refine design

### PR DESCRIPTION
**Pull request by hellangle**

### CODE REVIEW

File: src/JobCentral.Websites.Base/JobCentral.Websites.Base.Services/Cities/CityService.cs [hunk 1]
```diff
@@ -23,17 +23,17 @@ protected override System.Data.Entity.IDbSet<City> Table
...
                 query = query.Where(c => c.Label.ToLower() == specification.Label.ToLower());
             }
-            return query.OrderBy(c => c.Label);
+            return query.OrderBy(c => c.PostNumber);
         }
 
         public override int Count(CitySpecification specification)
         {
-            var query = Table.AsNoTracking().AsQueryable();
+            var query = Table.AsNoTracking().AsQueryable().Where(c => c.Deleted == null).Where(c => c.PostNumber.CompareTo("2000") > 0);
```
Comment (vgoloviznin) [inline src/JobCentral.Websites.Base/JobCentral.Websites.Base.Services/Cities/CityService.cs]: Just call the Get method here and count the results

File: src/JobCentral.Websites.Base/JobCentral.Websites.Base.Services/Cities/CityService.cs [hunk 2]
```diff
@@ -23,17 +23,17 @@ protected override System.Data.Entity.IDbSet<City> Table
 
         public override IQueryable<City> Get(CitySpecification specification)
         {
-            var query = Table.AsNoTracking().AsQueryable().Where(c => c.PostNumber.CompareTo("2000") > 0);
+            var query = Table.AsNoTracking().AsQueryable().Where(c => c.Deleted == null).Where(c => c.PostNumber.CompareTo("2000") > 0);
```
Comment (vgoloviznin) [inline src/JobCentral.Websites.Base/JobCentral.Websites.Base.Services/Cities/CityService.cs]: Combine both Where clauses into one

File: src/JobCentral.Websites.Base/JobCentral.Websites.Base.Services/Notifications/MandrillNotifierEmailer.cs [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Mandrill;
+
+namespace JobCentral.Websites.Base.Services.Notifications
+{
+    public class MandrillNotifierEmailer : INotifier, IEmailer
+    {
+        private readonly MandrillApi api;
```
Comment (vgoloviznin) [inline src/JobCentral.Websites.Base/JobCentral.Websites.Base.Services/Notifications/MandrillNotifierEmailer.cs]: Rename `api` to `_api`. Let's use default ReSharper code styles everywhere

File: src/JobCentral.Websites.Base/JobCentral.Websites.Base.Services/Notifications/MandrillNotifierEmailer.cs [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+        public MandrillNotifierEmailer()
+        {
+            this.api = new MandrillApi(ConfigurationManager.AppSettings["MandrillKey"]);
```
Comment (vgoloviznin) [inline src/JobCentral.Websites.Base/JobCentral.Websites.Base.Services/Notifications/MandrillNotifierEmailer.cs]: You don't need `this` here

File: src/JobCentral.Websites.Base/JobCentral.Websites.Base.Services/Notifications/MandrillNotifierEmailer.cs [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+        }
+
+        public void SendPasswordRecoveryEmail(string email, string link)
+        {
+            var message = new EmailMessage
+            {
+                to = new[] { new EmailAddress(email) },
+                from_email = "no-reply@jobcentral.dk",
```
Comment (vgoloviznin) [inline src/JobCentral.Websites.Base/JobCentral.Websites.Base.Services/Notifications/MandrillNotifierEmailer.cs]: Extract default values (from, subjects, templates names) into `web.config`

File: src/JobCentral.Websites.Base/JobCentral.Websites.Base.Services/Notifications/MandrillNotifierEmailer.cs [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+            message.AddGlobalVariable("QUESTIONLINK", link);
+
+            api.SendMessage(message, "PasswordRecoveryTemplate", null);
+        }
+
+        public void SendContactEmail(string name, string email, string comments)
+        {
+            var message = new EmailMessage
+            {
+                to = new[] { new EmailAddress("info@jobcentral.dk") },
```
Comment (vgoloviznin) [inline src/JobCentral.Websites.Base/JobCentral.Websites.Base.Services/Notifications/MandrillNotifierEmailer.cs]: Why do we need two different senders here?

File: src/JobCentral.Websites.Base/JobCentral.Websites.Base.Services/Users/UserService.cs [hunk 7]
```diff
@@ -10,9 +11,12 @@ namespace JobCentral.Websites.Base.Services.Users
     public class UserService : BaseService<User, UserSpecification>, IUserService
     {
         private readonly UserManager _userManager;
-        public UserService(IUnitOfWork context, UserManager userManager) : base(context)
+        private readonly IEmailer emailer;
+        public UserService(IUnitOfWork context, UserManager userManager, IEmailer emailer)
+            : base(context)
         {
             _userManager = userManager;
+            this.emailer = emailer;
```
Comment (vgoloviznin) [inline src/JobCentral.Websites.Base/JobCentral.Websites.Base.Services/Users/UserService.cs]: You don't need `this` here

File: src/JobCentral.Websites.Base/JobCentral.Websites.Base.Services/Users/UserService.cs [hunk 8]
```diff
@@ -61,5 +65,21 @@ public User GetById(string userId)
...
+                throw new Exception("Invalid email");
+            }
+
+            var code = _userManager.GeneratePasswordResetToken(user.Id);
+
+            var recoverUrl = string.Format(recoverLinkTemplate,
+                Uri.EscapeDataString(user.Id),
+                Uri.EscapeDataString(code));
+
+            this.emailer.SendPasswordRecoveryEmail(user.Email, recoverUrl);
```
Comment (vgoloviznin) [inline src/JobCentral.Websites.Base/JobCentral.Websites.Base.Services/Users/UserService.cs]: You don't need `this` here

File: src/JobCentral.Websites.Base/JobCentral.Websites.Base/Controllers/AccountController.cs [hunk 9]
```diff
@@ -85,5 +87,30 @@ public ActionResult Logout()
...
+        }
+
+        [HttpPost]
+        public virtual ActionResult RecoverPassword(string email)
+        {
+            var recoverUlrTemplate = string.Format("{0}?userId={{0}}&code={{1}}", ConfigurationManager.AppSettings["BaseUrl"]);
+
+            try
+            {
+                this._userService.RecoverPassword(email, recoverUlrTemplate);
```
Comment (vgoloviznin) [inline src/JobCentral.Websites.Base/JobCentral.Websites.Base/Controllers/AccountController.cs]: You don't need `this` here

File: src/JobCentral.Websites.Base/JobCentral.Websites.Base/Controllers/AccountController.cs [hunk 10]
```diff
@@ -85,5 +87,30 @@ public ActionResult Logout()
             AuthenticationManager.SignOut();
             return RedirectToAction("Index", "Home");
         }
+
+        [HttpPost]
```
Comment (vgoloviznin) [inline src/JobCentral.Websites.Base/JobCentral.Websites.Base/Controllers/AccountController.cs]: Why is this method decorated with `HttpPost`?

File: src/JobCentral.Websites.Base/JobCentral.Websites.Base/Controllers/AccountController.cs [hunk 11]
```diff
@@ -85,5 +87,30 @@ public ActionResult Logout()
...
+        public virtual ActionResult RecoverPassword(bool? success)
```
Comment (vgoloviznin) [inline src/JobCentral.Websites.Base/JobCentral.Websites.Base/Controllers/AccountController.cs]: You can use a default value for this variable `bool success = false`

File: src/JobCentral.Websites.Base/JobCentral.Websites.Base/Controllers/AccountController.cs [hunk 12]
```diff
@@ -85,5 +87,30 @@ public ActionResult Logout()
...
+        {
+            ViewBag.Success = success ?? false;
```
Comment (vgoloviznin) [inline src/JobCentral.Websites.Base/JobCentral.Websites.Base/Controllers/AccountController.cs]: Don't use `ViewBag` here, just pass success as a model to the view

File: src/JobCentral.Websites.Base/JobCentral.Websites.Base/Controllers/HomeController.cs [hunk 13]
```diff
@@ -4,20 +4,27 @@
...
 {
     public class HomeController : Controller
     {
-        public HomeController()
+        private readonly IEmailer emailer;
+        private readonly IJobAdvertService jobAdvertService;
+        public HomeController(IEmailer emailer, IJobAdvertService jobAdvertService)
         {
-            
+            this.emailer = emailer;
```
Comment (vgoloviznin) [inline src/JobCentral.Websites.Base/JobCentral.Websites.Base/Controllers/HomeController.cs]: You don't need `this` here

File: src/JobCentral.Websites.Base/JobCentral.Websites.Base/Controllers/HomeController.cs [hunk 14]
```diff
@@ -4,20 +4,27 @@
...
+            this.jobAdvertService = jobAdvertService;
         }
 
         public ActionResult Index()
         {
-            return View();
+            var model = new HomepageViewModel();
```
Comment (vgoloviznin) [inline src/JobCentral.Websites.Base/JobCentral.Websites.Base/Controllers/HomeController.cs]: Use an object initializer ` new HomepageViewModel {SomeField = SomeValue}`

File: src/JobCentral.Websites.Base/JobCentral.Websites.Base/Controllers/HomeController.cs [hunk 15]
```diff
@@ -36,7 +43,15 @@ public ActionResult Contact(ContactViewModel model)
         {
             if (!ModelState.IsValid) return View(model);
 
-            //TODO send email here
+            try
+            {
+                this.emailer.SendContactEmail(model.Name, model.Email, model.Comment);
```
Comment (vgoloviznin) [inline src/JobCentral.Websites.Base/JobCentral.Websites.Base/Controllers/HomeController.cs]: You don't need `this` here

File: src/JobCentral.Websites.Base/JobCentral.Websites.Base/Scripts/internals/infinity.scroll.js [hunk 16]
```diff
@@ -0,0 +1,70 @@
...
+        if (self.isLoading == true) return;
+
+        var $lastItem = self.$container.find('>div:last');
+        if ($lastItem.length == 0 || !$lastItem.offset) {
+            $(window).unbind('scroll');
+            return;
+        }
+        
+        
+        var shouldLoad = false;
```
Comment (vgoloviznin) [inline src/JobCentral.Websites.Base/JobCentral.Websites.Base/Scripts/internals/infinity.scroll.js]: You don't need an extra variable here, if the `if` statement returns false `return` from the function

File: src/JobCentral.Websites.Base/JobCentral.Websites.Base/Scripts/internals/infinity.scroll.js [hunk 17]
```diff
@@ -0,0 +1,70 @@
...
+                } else {
+                    var url = '';
+                    if (thiz.options.currentUrl.indexOf('?') < 0) {
+                        url = thiz.options.currentUrl + '?page=' + thiz.pageIndex;
+                    } else {
+                        url = thiz.options.currentUrl + '&page=' + thiz.pageIndex;
+                    }
+                    var state = self.options.state || { };
+                    state.pageIndex = thiz.pageIndex;
+                    history.pushState(state, 'page 2', url);
```
Comment (vgoloviznin) [inline src/JobCentral.Websites.Base/JobCentral.Websites.Base/Scripts/internals/infinity.scroll.js]: Why `page 2` is hardcoded here?

File: src/JobCentral.Websites.Base/JobCentral.Websites.Base/Scripts/internals/search.adverts.js [hunk 18]
```diff
@@ -260,4 +287,42 @@
         }
         self.loadAdverts(self.pageIndex);
     }
+
+    self.isLoading = false;
+    self.lazyLoadData = function() {
+        if (self.isLoading === true) return;
```
Comment (vgoloviznin) [inline src/JobCentral.Websites.Base/JobCentral.Websites.Base/Scripts/internals/search.adverts.js]: Why do you have this function here if you already have InfinityScroll module

Comment (hellangle) [inline src/JobCentral.Websites.Base/JobCentral.Websites.Base/Scripts/internals/search.adverts.js]: InfinityScroll module don't fit with the case

File: src/JobCentral.Websites.Base/JobCentral.Websites.Base/Scripts/internals/infinity.scroll.js (hunk 17 above)
Comment (hellangle) [inline src/JobCentral.Websites.Base/JobCentral.Websites.Base/Scripts/internals/infinity.scroll.js]: Currently we don't use it

File: src/JobCentral.Websites.Base/JobCentral.Websites.Base/Scripts/internals/infinity.scroll.js (hunk 16 above)
Comment (hellangle) [inline src/JobCentral.Websites.Base/JobCentral.Websites.Base/Scripts/internals/infinity.scroll.js]: Currently we don't use it

**Pull request merged**